linux/check_os_fullname/check_os_fullname.yml

```diff
@@ -58,33 +63,4 @@
           include_tasks: ../../common/test_rescue.yml
       always:
         - name: "Collect VMware Tools logs"
-          when:
-            - vmtools_log_dir is defined
-            - vmtools_log_dir
-          block:
-            - name: "Get VMware Tools log directory stat info"
-              include_tasks: ../utils/get_file_stat_info.yml
-              vars:
-                guest_file_path: "{{ vmtools_log_dir }}"
-
-            - name: "Collect VMware Tools logs"
-              when: guest_file_exists
-              block:
-                - name: "Archive VMware Tools debug logs"
-                  community.general.archive:
-                    path: "{{ vmtools_log_dir }}"
-                    dest: "{{ vmtools_log_dir }}.tgz"
-                    mode: "0644"
-                  delegate_to: "{{ vm_guest_ip }}"
-                  register: archive_vmtools_logs
-
-                - name: "Fetch VMware Tools debug logs"
-                  include_tasks: ../utils/fetch_file.yml
-                  vars:
-                    fetch_file_src_path: "{{ vmtools_log_dir }}.tgz"
-                    fetch_file_dst_path: "{{ current_test_log_folder }}/"
-                    fetch_file_ignore_errors: true
-                  when:
-                    - archive_vmtools_logs is defined
-                    - archive_vmtools_logs.changed is defined
-                    - archive_vmtools_logs.changed
+          include_tasks: ../utils/collect_vmtools_logs.yml
```

Comment:
How about log file `vmtools_log_dir`?

Reply:
`vmtools_log_dir` is the directory for saving `vmtools_vmtoolsd_log_file` and `vmtools_vmsvc_log_file` in `linux/utils/enable_vmtools_logging.yml`. Using `vmtools_log_dir` needs to archive the directory to collect the two logs. Since we know the paths of `vmtools_vmtoolsd_log_file` and `vmtools_vmsvc_log_file`, we can directly collect the two log files.

Reply:
If it's only used in that task file then they can be moved to a block or via `include_tasks` vars.